src/dvc_data/hashfile/hash.py

```diff
@@ -145,16 +145,16 @@ def __init__(self, *args, **kwargs):
         kwargs.setdefault("bytes", True)
         super().__init__(*args, **kwargs)
         self._logged = False
+        self.fname = kwargs.get("desc", "")
```

Comment:
depending on whether this is actually mandatory...

```suggestion
        self.fname = kwargs["desc"]
```